File: 0228-summary-ranges/0228-summary-ranges.py
class Solution:
    def summaryRanges(self, nums: list[int]) -> list[str]:
        result = []
        i = 0
        n = len(nums)

        while i < n:
            start = nums[i]

            # Shift the inner boundary index step forward
            while i + 1 < n and nums[i + 1] == nums[i] + 1:
                i += 1

            # Format strings using efficient Python f-strings
            if start == nums[i]:
                result.append(str(start))
            else:
                result.append(f"{start}->{nums[i]}")

            i += 1

        return result

File: 0228-summary-ranges/0228-summary-ranges.js
/**
 * @param {number[]} nums
 * @return {string[]}
 */
var summaryRanges = function(nums) {
    const result = [];
    let i = 0;

    while (i < nums.length) {
        const start = nums[i];

        // Track continuous consecutive integer jumps
        while (i + 1 < nums.length && nums[i + 1] === nums[i] + 1) {
            i++;
        }

        // Format single elements or interval blocks cleanly
        if (start === nums[i]) {
            result.push(String(start));
        } else {
            result.push(start + "->" + nums[i]);
        }

        i++;
    }

    return result;
};
